### ISSUE TITLE
imp deprecated

### ISSUE THREAD

**Issue by tschm** [cvxpy/cvxpy#383]
/opt/conda/lib/python3.6/site-packages/CVXcanon.py:15: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp

**Comment by rAntonioh** [comment]
is there any update to this issue? it is at the root of a bug whose error starts as:
ImportError: /lib64/libc.so.6: version `GLIBC_2.14' not found 
the common fix is to do conda update libgcc, however this does not work on my linux system
when looking at the error trace, i find that the problem is when importing CVXcanon...

it is using imp.py which says it is deprecated, perhaps this is connected?

**Comment by rileyjmurray** [comment]
To clarify, are you running 1.0 or 0.4?

We're about to switch to 1.0, so we're really interested in making installation / deployment painless in 1.0.

**Comment by SteveDiamond** [comment]
Closing due to lack of response.

**Comment by wkschwartz** [comment]
I can verify this problem.

```Python session
$ python -Wonce
Python 3.7.1 (default, Nov  6 2018, 18:45:35) 
[Clang 10.0.0 (clang-1000.11.45.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from cvxpy import __version__; print(__version__)
/Users/billy/Documents/assigner/venv/lib/python3.7/site-packages/cvxpy/cvxcore/python/cvxcore.py:15: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/importlib/_bootstrap.py:219: ImportWarning: can't resolve package from __spec__ or __package__, falling back on __name__ and __path__
  return f(*args, **kwds)
1.0.10
```

**Comment by wkschwartz** [comment]
I can verify this problem.
```Python session
$ python -Wonce
Python 3.7.1 (default, Nov  6 2018, 18:45:35) 
[Clang 10.0.0 (clang-1000.11.45.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from cvxpy import __version__; print(__version__)
/.../venv/lib/python3.7/site-packages/cvxpy/cvxcore/python/cvxcore.py:15: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
/usr/local/Cellar/python/3.7.1/Frameworks/Python.framework/Versions/3.7/lib/python3.7/importlib/_bootstrap.py:219: ImportWarning: can't resolve package from __spec__ or __package__, falling back on __name__ and __path__
  return f(*args, **kwds)
1.0.10
```

**Comment by wkschwartz** [comment]
The problem is here:

https://github.com/cvxgrp/cvxpy/blob/8ceacfec7dc2c8c5d26eb49973fdb3d1938cf835/cvxpy/cvxcore/python/cvxcore.py#L15

**Comment by christian-oudard** [comment]
I am also having this problem, and I would also like to find a resolution to this. Can we reopen it?

**Comment by wkschwartz** [comment]
Look like SWIG 4.0 will fix the problem: https://github.com/swig/swig/issues/1621#issuecomment-527250111

**Comment by SteveDiamond** [comment]
Fixed in 1.1.
